src/node_api.cc

```diff
@@ -629,38 +647,45 @@ const char* error_messages[] = {nullptr,
                                 "Unknown failure",
                                 "An exception is pending"};
 
-void napi_clear_last_error() {
-  static_last_error.error_code = napi_ok;
+void napi_clear_last_error(napi_env env) {
+  env->last_error.error_code = napi_ok;
 
   // TODO(boingoing): Should this be a callback?
-  static_last_error.engine_error_code = 0;
-  static_last_error.engine_reserved = nullptr;
+  env->last_error.engine_error_code = 0;
+  env->last_error.engine_reserved = nullptr;
 }
 
-napi_status napi_set_last_error(napi_status error_code,
-                                uint32_t engine_error_code = 0,
-                                void* engine_reserved = nullptr) {
-  static_last_error.error_code = error_code;
-  static_last_error.engine_error_code = engine_error_code;
-  static_last_error.engine_reserved = engine_reserved;
+napi_status napi_set_last_error(napi_env env, napi_status error_code,
+                                uint32_t engine_error_code,
+                                void* engine_reserved) {
+
+  if (env == nullptr) {
+    if (error_code == napi_ok) {
```

Comment:
I'd agree that we should never be calling this with napi_ok, and it is an internal method so that would be a bug on our side.    Maybe an assert instead ?